pkg/datapath/linux/probes/probes.go

```diff
@@ -201,6 +201,23 @@ func (p *ProbeManager) SystemKernelHz() (int, error) {
 // returns an error when parameters required by Cilium are not enabled. It logs
 // warnings when optional parameters are not enabled.
 func (p *ProbeManager) SystemConfigProbes() error {
+	requiredParams := p.GetRequiredConfig()
+	for param, status := range requiredParams {
+		var canBeModule string
+		switch param {
+		case "CONFIG_NET_SCH_INGRESS", "CONFIG_NET_CLS_BPF":
+			canBeModule = " (or module)"
+		}
+		if !status {
+			return fmt.Errorf("%s kernel parameter%s is required", param, canBeModule)
+		}
+	}
+	optionalParams := p.GetOptionalConfig()
+	for param, status := range optionalParams {
+		if !status {
+			fmt.Printf("%s kernel parameter is required", param)
```

Comment:
Please use `log.Warning()` as was the case before. And update the message, these parameters are not required.

Also now that we have this processing based on `Get*Config()`, we have to remove the previous content of `SystemConfigProbes()` function, right? We don't want to probe each parameter twice in the function.